File: arrays/11. Container With Most Water/maxArea.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        amount = 0
        i, j = 0, len(height) - 1

        while i < j:
            tmp_amount = min(height[i], height[j]) * (j - i)
            # Keep track of the biggest amount
            amount = max(amount, tmp_amount)

            # Move the pointer with the smaller height
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        
        return amount

File: arrays/11. Container With Most Water/maxArea.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
  let left = 0;
  let right = height.length - 1;
  let area = 0;

  while (left < right) {
    let currentArea = (right - left) * Math.min(height[left], height[right]);
    area = Math.max(currentArea, area);

    if (height[left] <= height[right]) left++;
    else right--;
  }

  return area;
};